app/code/Magento/Quote/Model/Quote/Address/Total.php

```diff
@@ -54,6 +55,9 @@ public function __construct(
      */
     public function setTotalAmount($code, $amount)
     {
+        /* (Fixes issue #18027) Round the total amount to 4 decimal places, to avoid floating point overflows */
+        $amount = is_float($amount) ? (float)number_format($amount, 4) : $amount;
```

Comment:
I think this is why the tests fail. number_format and then casting to float changed the value apparently

Reply:
Yes but what is the solution then? Rewrite the test cases?

Reply:
No, rewrite the solution. Problem is the following: If the amount is 5084.65, number_format(5084.65, 4) returns 5,084.6500, which, if casted to float, results in a value of 5, which is obviously wrong. Using number_format($amount, 4, '.', '') would work, but I'm not sure it solves the issue (as it returns what was put in). What exactly are you trying to accomplish?

Reply:
I wanted to round the float to 4 decimal places. `number_format($amount, 4, '.', '')` would work perfectly. I completely forgot that `number_format` takes `,` as the default thousand separator. The problem is, in certain situations, due to floating point overflow, the total is coming as `1.0E-45` instead of 0. So rounding it off would solve the issue

Reply:
As far as I know, number_format does no rounding at all, it just formats the float as a string according to the specified separator symbols. Maybe you wanted to use round($amount, 4)? This would also not need an explicit cast, as it already returns the rounded float value.

Reply:
You're right. I should probably use `round` instead of `number_format`. Thanks for the suggestion